### PR TITLE
[dmd-cxx] Backport a number of bindings and fixes from master

### PR DESCRIPTION
**Pull request by ibuclaw**
Bindings are HPPA support, and some work towards OSX.

Backport of core.internal.hash is a primer for another PR that will fix hashing of complex reals.

### CODE REVIEW

File: src/core/sys/posix/aio.d [hunk 1]
```diff
@@ -63,6 +63,19 @@ version (CRuntime_Glibc)
         }
     }
 }
+else version (OSX)
```
Comment (jacob-carlborg) [inline src/core/sys/posix/aio.d]: Other modules are using `Darwin`: https://github.com/dlang/druntime/blob/59607879d66de79e2fa159048ae74e52c7876ea6/src/core/sys/posix/dirent.d#L21-L28.

Comment (ibuclaw) [reply src/core/sys/posix/aio.d]: @jacob-carlborg - That needs fixing in master too right?

Comment (jacob-carlborg) [reply src/core/sys/posix/aio.d]: Yes, I didn't notice this PR targeted `dmd-cxx`.

Comment (ibuclaw) [reply src/core/sys/posix/aio.d]: #2470.

**Comment by ZombineDev** [comment]
@ibuclaw the only commit that I have doubts for is 0cbe48dc83fdfe2fb169c49563ed1502faa299c9. What was your process in making it, as I guess it encompasses tens of commits from the master branch.

**Comment by ibuclaw** [comment]
@ZombineDev - meld mostly.

The ultimate patch is it intended for is https://github.com/dlang/druntime/pull/2356

**Comment by ZombineDev** [comment]
Alright, thanks for the info!

**Pull request merged**